pytket/extensions/qiskit/backends/aer.py

```diff
@@ -660,6 +663,71 @@ def __init__(self, n_qubits: int = 40) -> None:
         ]
 
 
+class AerDensityMatrixBackend(_AerBaseBackend):
+    """
+    Backend for running simulations on the Qiskit Aer density matrix simulator.
+
+    :param noise_model: Noise model to apply during simulation. Defaults to None.
+    :param n_qubits: The maximum number of qubits supported by the backend.
+    """
+
+    _supports_density_matrix = True
+    _supports_state = False
+    _memory = False
+    _noise_model = None
+    _needs_transpile = True
+    _supports_expectation = True
+
+    _qiskit_backend_name = "aer_simulator_density_matrix"
+
+    _allowed_special_gates = {
+        OpType.Measure,
+        OpType.Barrier,
+        OpType.Reset,
+        OpType.RangePredicate,
+    }
+
+    def __init__(
+        self,
+        noise_model: Optional[NoiseModel] = None,
+        n_qubits: int = 40,
+    ) -> None:
+        super().__init__()
+        self._qiskit_backend = qiskit_aer_backend(self._qiskit_backend_name)
+        self._noise_model = noise_model
+        gate_set = _tket_gate_set_from_qiskit_backend(self._qiskit_backend).union(
```

Comment:
type for this would be great

Reply:
added in https://github.com/CQCL/pytket-qiskit/pull/380/commits/1d5fd8c8871e7393745416e91c933d2d29bbd7ad